lib/collection.js

```diff
@@ -128,7 +129,7 @@ function Collection(db, topology, dbName, name, pkFactory, options) {
   const namespace = new MongoDBNamespace(dbName, name);
 
   // Get the promiseLibrary
-  const promiseLibrary = options.promiseLibrary || Promise;
+  const promiseLibrary = PromiseProvider.get(options, db, topology);
```

Comment:
Here's a case that I'm taking about. It's a little more comprehensive then it previously was.